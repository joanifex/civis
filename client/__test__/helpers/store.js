import configureMockStore from 'redux-mock-store';


const store = (model) => {
  const mockStore = configureMockStore();
  return mockStore(model)
}

export default store;
