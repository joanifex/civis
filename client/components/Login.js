import React from 'react';
import { handleLogin } from '../actions/auth';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import { civisBlue } from './styles.scss';

class Login extends React.Component {

  handleSubmit = (e) => {
    e.preventDefault();
    const email = this.refs.email.value;
    const password = this.refs.password.value;
    this.props.dispatch(handleLogin(email, password));
  }

  render() {
    return(
      <div className="row">
        <div className="col s12 m10 offset-m1 l8 offset-l2">
          <div className="card grey lighten-4">
            <div className="card-content black-text">
              <span className="card-title">Login</span>
                <form onSubmit={ this.handleSubmit }>
                  <div className="input-field">
                    <i className="fa fa-envelope-o prefix"></i>
                    <input id='email' ref='email' className="icon_prefix" type='email' required placeholder='Email' />
                  </div>
                  <div className="input-field">
                    <i className="fa fa-unlock-alt prefix"></i>
                    <input ref='password' className="icon_prefix" type='password' required placeholder='Password' />
                  </div>
                  <input type='submit' className={` waves-effect waves-light btn ${civisBlue}`} />
                  <Link to='/sign_up' className='waves-effect waves-teal btn-flat right'>
                    Sign Up
                  </Link>
                </form>
            </div>
          </div>
        </div>
      </div>

    );
  }
}

export default connect()(Login);
