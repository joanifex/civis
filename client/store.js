import { createStore, compose, applyMiddleware } from 'redux';
import { syncHistoryWithStore } from 'react-router-redux';
import { browserHistory } from 'react-router';
import thunk from 'redux-thunk';

import rootReducer from './reducers/index';

const enhancers = compose(
  applyMiddleware(thunk),
  window.devToolsExtension ? window.devToolsExtension() : f => f
)

let appDataSet = document.getElementById('app').dataset;
let remember = appDataSet.remember;

remember = remember === 'true' ? true : false;

const store = createStore(rootReducer, { auth: { isAuthenticated: remember } }, enhancers)

if(module.hot) {
  module.hot.accept('./reducers/',() => {
    const nextRootReducer = require('./reducers/index').default;
    store.replaceReducer(nextRootReducer);
  });
}

export const history = syncHistoryWithStore(browserHistory, store)

export default store
