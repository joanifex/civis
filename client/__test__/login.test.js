import React from 'react';
import Login from '../components/Login.js';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './helpers/store';

test('Component is rendered', () => {
  const component = renderer.create(
    <Provider store={store({})}>
      <Login />
   </Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
