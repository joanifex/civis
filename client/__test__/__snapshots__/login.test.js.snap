// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is rendered 1`] = `
<div
  className="row"
>
  <div
    className="col s12 m10 offset-m1 l8 offset-l2"
  >
    <div
      className="card grey lighten-4"
    >
      <div
        className="card-content black-text"
      >
        <span
          className="card-title"
        >
          Login
        </span>
        <form
          onSubmit={[Function]}
        >
          <div
            className="input-field"
          >
            <i
              className="fa fa-envelope-o prefix"
            />
            <input
              className="icon_prefix"
              id="email"
              placeholder="Email"
              required={true}
              type="email"
            />
          </div>
          <div
            className="input-field"
          >
            <i
              className="fa fa-unlock-alt prefix"
            />
            <input
              className="icon_prefix"
              placeholder="Password"
              required={true}
              type="password"
            />
          </div>
          <input
            className="btn"
            type="submit"
          />
          <a
            className="btn grey"
            onClick={[Function]}
            style={Object {}}
          >
            Sign Up
          </a>
        </form>
      </div>
    </div>
  </div>
</div>
`;
